pkg/translate/translate_value.go

```diff
@@ -289,6 +288,46 @@ func translateHPASpec(inPath string, outPath string, value interface{}, valueTre
 	if _, err := tpath.DeleteFromTree(valueTree, util.ToYAMLPath(maxPath), util.ToYAMLPath(maxPath)); err != nil {
 		return err
 	}
+
+	cuPath := newPS + ".cpu.targetAverageUtilization"
+	cuVal, found, err := name.GetFromTreePath(valueTree, util.ToYAMLPath(cuPath))
```

Comment:
the operation here should probably be in a helper function - it's too detailed to follow at this level.

Reply:
I think the translateHPASpec function is already helper function to handle special translation, and since each hpa setting has slightly different handling so is not easy to extract general ways from it. So I try to refactor the logic inside this function.